### DISCUSSION TITLE
llama-3.1-sonar-small-128k-online not using web look-up

### DISCUSSION THREAD

**Discussion by AlKamber** [perplexityai/api-discussion#69] [Q&A]
I cannot get llama-3.1-sonar-small-128k-online to perform web look-ups, and it always produces results stating it is "unable to search the internet in real-time or access specific company databases directly", is it possible to fix?

         `response = self.client.chat.completions.create(
                model="llama-3.1-sonar-small-128k-online",
                messages=[
                    {
                        "role": "system",
                        "content": system_prompt
                    },
                    {
                        "role": "user",
                        "content": prompt
                    }
                ],
                temperature=0.3,
                max_tokens=1000
            )`

`def __init__(self, api_key: str):
        self.client = OpenAI(api_key=api_key, base_url="https://api.perplexity.ai")`

It is fairly disappointing compared to what the UI offers, although I understand the two not aligning as explained in the FAQ, I'd still expect the API to perform web look-ups to retrieve information.

**Comment by shubhang98** [comment]
What is your prompt? This should not be happening.
